src/zenml/zen_server/routers/models_endpoints.py

```diff
@@ -250,64 +232,148 @@ def update_model_version(
     )
 
 
-######################
-# Model Version Links
-######################
+@router.delete(
+    "/{model_name_or_id}" + MODEL_VERSIONS + "/{model_version_name_or_id}",
+    responses={401: error_response, 404: error_response, 422: error_response},
+)
+@handle_exceptions
+def delete_model_version(
+    model_name_or_id: Union[str, UUID],
+    model_version_name_or_id: Union[str, UUID],
+    _: AuthContext = Security(authorize, scopes=[PermissionType.WRITE]),
+) -> None:
+    """Delete a model by name or ID.
+
+    Args:
+        model_name_or_id: The name or ID of the model containing version.
+        model_version_name_or_id: The name or ID of the model version to delete.
+    """
+    zen_store().delete_model_version(
+        model_name_or_id, model_version_name_or_id
+    )
+
+
+##########################
+# Model Version Artifacts
+##########################
 
 
 @router.get(
```

Comment:
Actually why do these link endpoints need to exist? When would a user need to query this? Aren't those links/IDs already included in the model version models?

Reply:
Was talking with @znegrin about the linking in the product sync just now as well. Ended up also concluding that maybe we don't need so much exposure of the linking functionality to the user after all (as I'd originally thought on reading this PR)

Reply:
Noted